src/app/reducers/incidents/incidentSlice.js

```diff
@@ -62,6 +62,55 @@ export const errorLog = state => state.incidents.meta.errorLog;
 export const refreshState = state => state.incidents.meta.refresh;
 // For getting the value of the current state throughout the app;
 
+
+// Formatter Function for preparing selectedIncident Data for DetailsBlock
+// This moves the responsibility out of the component, and into the reducer.
+// Transformation and formatting of data makes more sense in reducer middleware than the React component. (eg: Smart vs Dumb components)
+export const getDetailBlocks = (state) => {
+    const selectedId = selectIncident(state); // Reusing selectors above
+    const incidentList = listIncidents(state); // Reusing selectors above
+    const selectedIncident = incidentList.find((incident) => !incident._id.indexOf(selectedId));
+
+    if (!selectedIncident) { return {}; }
+
+    const { geographics, incident } = selectedIncident;
+
+    // Set up the data representation of our detail blocks.
+    // This keeps the data formatting responsibility separate from presentational components.
+    // We can break this down further if we wanted to with each "block" having its own formatter function outputting the correct object structure, then adding it to the array.
+    const blocks =  [
+        {
+            title: 'Incident Information',
+            rows:  [
+                { type: 'Incident', content: geographics.municipal },
+                { type: 'State', content: geographics.state }, 
+                { type: 'Region', content: geographics.region }, 
+                { type: 'ID', content: selectedIncident._id.substr(selectedId.length - 5) }, 
+                { type: 'Initial Time', content: `${geographics.time_stamp} ${geographics.time_zone} `}, 
+                { type: 'Zipcodes Affected', content: incident.zip_codes.map(zip => `${zip}, `) }
+            ] 
+        },
```

Comment:
This data block represents a DetailsBlock component. A useful habit to cultivate in your dev career is learning to look at a flat UI design and start thinking about it as if it were only JSON -- this makes planning out React props much easier!

You start to recognize data patterns in designs and can build cleaner components when you know what the data should look like.